core/src/main/java/com/datastax/oss/driver/internal/core/util/Dependency.java

```diff
@@ -15,7 +15,9 @@
  */
 package com.datastax.oss.driver.internal.core.util;
 
-import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
```

Comment:
Shouldn't we add a comment somewhere to warn future maintainers that this class should be extra careful when importing anything outside of the standard library?

Reply:
So... I'm a little torn on this, frankly.

I'd feel better about saying something like this if I understood more cleanly _why_ the change in question resulted in the behaviour we saw.  But it seems to be the case that introducing a dependency upon a shaded class somehow:

* caused Graal to eval this class as a runtime class
* only did so for Graal 22.2; the original code has always worked fine on 22.2

I... just don't know if I have enough there to extract some kind of meaningful guidance.  We've already seen that changes between Graal versions can have odd impacts on what was perfectly valid code.  This seems like another one of those, but I just don't see a way to draw a lesson from that which can be clearly stated and apply to future use.

I'm open to suggestions if anybody has any.

Reply:
Yeah pretty hard to anticipate this kind of behavior change, I would vote for no specific comment but up to you.